### PR TITLE
feat: implement initial transaction validation

### LINKED ISSUE #350: Validate transactions before executing
Here's an example on how: https://github.com/lambdaclass/ethereum_rust/pull/135

### PR DESCRIPTION
**Pull request by TomasArrachea**
Closes #350 

This PR implements the initial check of the transaction. Consume the gas for creation of the transaction.

### CODE REVIEW

File: tests/evm.rs [hunk 1]
```diff
@@ -467,7 +467,7 @@ fn test_calldatacopy_memory_offset() {
     let program = Program::from(operations);
     let mut env = Env::default();
     env.tx.data = Bytes::from(vec![0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
-    env.tx.gas_limit = 1000;
+    env.tx.gas_limit = 1000 + gas_cost::TX_BASE_COST;
```
Comment (JulianVentura) [inline tests/evm.rs:470]: Same

File: tests/evm.rs [hunk 2]
```diff
@@ -530,7 +530,7 @@ fn test_calldatacopy_calldataoffset_bigger_than_calldatasize() {
...
```
Comment (JulianVentura) [inline tests/evm.rs:533]: Same

**Review by JulianVentura — CHANGES_REQUESTED** [review]
Some minor changes on tests, the rest looks good to me.

**Review by JulianVentura — APPROVED** [review]

**Review by MegaRedHand — APPROVED** [review]
LGTM

**Pull request merged**